.tool/lint

```diff
@@ -4,6 +4,12 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+
+# Run pre-commit
+PYTHON=$(command -v python python3|head -n1)
+which pre-commit >/dev/null || $PYTHON -m pip install --user pre-commit
```

Comment:
I'm really uncomfortable with running `pip` as part of a lint step; and I suspect that not all developers will welcome this. Could I recommend instead that you follow the convention used immediately below, i.e. erroring (or perhaps just warning) with instructions on how to install pre-commit?

Reply:
I moved the install code to Makefile following the pattern used with other tools.

Reply:
I did not explain myself correctly. What I meant is that automatically installing a Python tool via `pip` violates POLA, the Principle Of Least Astonishment. The make process pulls in Go modules - a Go developer will not be surprised by this. Installing Python stuff under a home directory, in non-Go places, that is unexpected. For some people, undesired.

I would suggest that the installation be done manually, a deliberate action on the part of the developer. I would suggest that if the tool is missing, you emit a warning or error with appropriate instructions.

Reply:
See my suggestion below, in comment I made about the Makefile.  We address this concern and make the thing run faster by only installing it once at image-build time instead of PR testing time.